### ISSUE TITLE
Is it possible to write to the local file or sent get/post requests via Scripting?

### ISSUE THREAD

**Issue by amorphius** [ProxymanApp/Proxyman#745]
I want to use Proxyman as middleman and pass some intercepted data to another application. So I need somehow to communicate with it. Either via local file or with simple http request. Is it possible?

**Comment by NghiaTranUIT** [comment]
Maybe it's possible by using the Scripting tool 🤔 

You can use JS to write to flow `onRequest` or `onResponse` to a file (This feature has not implemented yet, it's just an idea)

**Comment by NghiaTranUIT** [comment]
You can checkout this ticket: #741 because it might be related

**Comment by amorphius** [comment]
Yes, this seems related. And is it possible to make http request? Similar to `fetch`

**Comment by NghiaTranUIT** [comment]
at the moment, it's not possible because the script runs synchronize, not async (Promise doesn't fully support). If you can make a `fetch` works synchronize, I suppose that it should work

**Comment by NghiaTranUIT** [comment]
Write to file is easier, so I can support it if you like 😄

**Comment by amorphius** [comment]
That would be great 👍👍👍

**Comment by amorphius** [comment]
BTW how can I use synchronous fetch? How to import it into my script?

**Comment by NghiaTranUIT** [comment]
Good question. Since JavascriptCore from Apple doesn't have `fetch` by default, we have to find some way to add it as an addon. I don't think that it's easy.

I will support the "Write to file" and send you a beta

For a fetch, I will research and support it soon 👍

**Comment by NghiaTranUIT** [comment]
@amorphius please check out this build at https://github.com/ProxymanApp/Proxyman/issues/741#issuecomment-752842805 👍

**Comment by amorphius** [comment]
thanks!
